ex2/ex2.c

```diff
@@ -1,14 +1,22 @@
-// Write a program that opens the text.txt  file (with the `fopen()` system call) located in this directory 
-// and then calls `fork()` to create a new process. Can both the child and parent access the file descriptor 
+// Write a program that opens the text.txt  file (with the `fopen()` system call) located in this directory
+// and then calls `fork()` to create a new process. Can both the child and parent access the file descriptor
 // returned by `fopen()`? What happens when they are written to the file concurrently?
 
+// > yes, they can both access the file concurrently
+// > the text is added one after the other
+
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
 
 int main(void)
 {
-    // Your code here 
-    
+    // Your code here
+    FILE *file = fopen("./text.txt", "a");
```

Comment:
This interesting, why the choice to use the append ("a") option? It's fine, but it means every time you run the file, you add a new "text to be written over" string to your text.txt file.

Reply:
Maybe there's a different assumption I should have as a programmer, but as a user of files (that already exist) I typically add to the contents of a file or modify it, rather than replacing it or starting fresh each time. I used that approach here. (Though I guess my initial guess at what would happen ("text to be written over") isn't a good look if I'm defining the behavior of the fopen command this way.)